utils/window/window.go

```diff
@@ -86,10 +81,6 @@ func (w *window[T]) Oldest() (T, bool) {
 
 // Length returns the number of elements in the window.
 func (w *window[T]) Length() int {
-	w.lock.Lock()
-	defer w.lock.Unlock()
```

Comment:
The function does not modify the content. The lock was previously used because `Oldest` and `Length` were potentially removing content. It was to prevent concurrent update.

Reply:
Though `Length` and `Oldest` don't update any values it's possible that another thread will call `Add` which will result in `Length` and `Oldest` having race-y reads

Reply:
I see 👍

Reply:
A change has been pushed 👍  Sorry I thought that this kind of concurrent write/read in our use-case was meaningless.